federation/pkg/federatedtypes/adapter.go

```diff
@@ -56,6 +56,14 @@ type FederatedTypeAdapter interface {
 	NewTestObject(namespace string) pkgruntime.Object
 }
 
+// SchedulerAdapter defines additional operations for those controllers which need
+// additional scheduling logic.
+type SchedulerAdapter interface {
```

Comment:
Its not necessary, but seems logically apt to me. Also, I guess, if the adapter is typecasted to an interface with only 1 method (Schedule), then the other adapter methods would probably not be available on it within the method call (in case a need that there might be,.... )